modules/tinymce/src/core/main/ts/selection/DetailsElement.ts

```diff
@@ -35,9 +36,25 @@ const filterDetails = (editor: Editor): void => {
   });
 };
 
+const preventSplittingSummary = (editor: Editor): void => {
+  editor.on('NodeChange', () => {
```

Comment:
This would execute way to often better to run this normalization step after the insertContent operation stick it in the `InsertContentImpl` module.

Reply:
Moved! And fixed the function a bit to work with nested details elements.